public/content/en/vibed/deploy-on-heroku.md

```diff
@@ -0,0 +1,183 @@
+# Deploy on heroku
+
+### Pre-requirements
+
+- You must have an [account]((https://signup.heroku.com/login)) on heroku
+- You should have [git](https://git-scm.com/) installed
+- Your application should compile without error using dmd (`dub build --build=release`). 
+
+
+### 1 : Setup the app
+
+The first thing we need to do before deploying our app to the cloud is bind our app's port to heroku's one. 
+Heroku sets the PORT variable that you are supposed to bind, and listens on tcp/80.
+
+
+```d
+shared static this() {
+  // ...
+  auto settings = new HTTPServerSettings;
+  // Provide a default port in case of the $PORT variable isn't set.  
+  settings.port = environment.get("PORT", "8080").to!ushort;
+  listenHTTP(settings, router);
+}
+```
+
+You also need to create a Procfile, which is a text file in the root directory of your application, responsible to explicitly declare what command 
+should be executed to start your app.
+
+The Procfile in the example app looks like this:
+
+```
+web: ./hello-world
+```
+
+### 2 : Prepare the app 
+
+Before going further you should login to the heroku cli.
+
+To login you have to use the [heroku toolbelt](https://toolbelt.heroku.com/standalone).
+
+This provides you access to the Heroku Command Line Interface (CLI), which can be used for managing and scaling your applications and add-ons.
+
+After installing the toolbet just run 
+
+```
+$ heroku login
+```
+
+### 3 : Create the app 
+
+To do so you can go on the [heroku dashboard](https://dashboard.heroku.com) and create a new app. 
+After doing so memorize the name of your app, you will use it later. 
+
+
+
+or use the command line like this 
+
+```
+$ heroku create
+Creating app... done, ⬢ rocky-hamlet-67506
+https://rocky-hamlet-67506.herokuapp.com/ | https://git.heroku.com/rocky-hamlet-67506.git
+```
+
+The app's name here is rocky-hamlet-67506. 
+
+### Deploy using git 
+
+To deploy your app you will use git and push your code to heroku as if you were pushing on your git repo. 
```

Comment:
You can deploy your app directly from `git` - it will be a separate git `remote` endpoint to which new releases can be pushed.